### PR TITLE
platform: imx: Enable IRQ_STEER driver

### PR DESCRIPTION
**Pull request by paulstelian97**
In #1747 I forgot to actually call the platform_interrupt_init() function 
where I register the IRQ_STEER controller. This patch fixes that and activates
the IRQ_STEER controller.

(also, this passed the smoke test, namely I am in fact getting correctly numbered interrupts from the WIP code on the ESAI and EDMA drivers)

Signed-off-by: Paul Olaru <paul.olaru@nxp.com>

### CODE REVIEW

**Comment by paulstelian97** [comment]
@tlauda Please put it on hold, we need to do more testing.

File: src/platform/imx8/platform.c [hunk 1]
```diff
@@ -156,6 +156,9 @@ int platform_init(struct sof *sof)
 
 	clock_set_freq(CLK_CPU(cpu_get_id()), CLK_MAX_CPU_HZ);
 
+	/* init IRQ_STEER */
+	platform_interrupt_init();
+
 	/* init DMA */
```
Comment (dbaluta) [inline src/platform/imx8/platform.c]: @paulstelian97 previous patch is pretty vague for the average reviewer :D. Also, please make the commits impersonal. 
So, the commit message should look like this:

Initial implementation of platform code did not initialize the platform interrupt controller.
Fix this now by calling platform_interrupt_init() .

File: src/drivers/imx/interrupt.c [hunk 2]
```diff
@@ -241,7 +241,7 @@ static inline void handle_irq_batch(struct irq_cascade_desc *cascade,
 	while (status) {
```
Comment (dbaluta) [inline src/drivers/imx/interrupt.c]: That's a nasty bug indeed. Please use drivers: interrupt: in the subject.

**Comment by paulstelian97** [comment]
Alright, @tlauda It's now ready and fully tested, this can now be merged. Re-requested review because of extra bugfix.

**Pull request merged**